<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Usage • singlecellviz</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Usage">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">singlecellviz</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/usage.html">Usage</a></li>
    <li><a class="dropdown-item" href="../articles/database.html">Database</a></li>
    <li><a class="dropdown-item" href="../articles/development.html">Development</a></li>
    <li><a class="dropdown-item" href="../articles/deployment.html">Deployment</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/metzger-chambon/singlecellviz/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Usage</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/metzger-chambon/singlecellviz/blob/master/vignettes/usage.Rmd" class="external-link"><code>vignettes/usage.Rmd</code></a></small>
      <div class="d-none name"><code>usage.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>SingleCellViz is a RShiny App with the aim to quickly and
dynamically:</p>
<ul>
<li>visualize</li>
<li>explore</li>
<li>highlight</li>
<li>download</li>
</ul>
<p>results of scRNA-seq datasets.</p>
<p>We believe that giving an <strong>easy access</strong> through a
RShiny App will enable a better <strong>visibility</strong> of the
studies, and maximize the use of already available datasets.</p>
<p>SingleCellViz is not meant to perform new analysis, but to guide
through already analyzed datasets, by enabling easy exploration and
presenting important results.</p>
<p>SingleCellViz is structured in different tabs that will be described
in the following chapters of this document. Except for the homepage tab,
which is general, all the other tabs are dependent on the dataset
selected. That means that the structure of one tab is always the same,
but its content (the data), is <strong>reactive to the choice of the
study</strong>.</p>
</div>
<div class="section level2">
<h2 id="homepage-tab">Homepage tab<a class="anchor" aria-label="anchor" href="#homepage-tab"></a>
</h2>
<p>The homepage presents the aim of the app and how to use it.</p>
</div>
<div class="section level2">
<h2 id="select-a-dataset">Select a dataset<a class="anchor" aria-label="anchor" href="#select-a-dataset"></a>
</h2>
<p>Selecting a dataset is pretty straight-forward. You have to click on
the drop-down menu on the left, under <code>Select a dataset:</code>,
and click on the study of interest. You can change it at any time, no
matter the tab you currently are on.</p>
</div>
<div class="section level2">
<h2 id="general-information-tab">General information tab<a class="anchor" aria-label="anchor" href="#general-information-tab"></a>
</h2>
<p>The general information tab contains a box with the:</p>
<ul>
<li>title</li>
<li>description</li>
<li>date</li>
<li>doi (if relevant)</li>
</ul>
<p>of the selected dataset.</p>
<p>On the right, some <strong>basics statistics</strong> are
highlighted:</p>
<ul>
<li>number of cells</li>
<li>number of features</li>
<li>number of samples</li>
</ul>
</div>
<div class="section level2">
<h2 id="explore-tab">Explore tab<a class="anchor" aria-label="anchor" href="#explore-tab"></a>
</h2>
<p>This is where SingleCellViz gets interesting! Two parameters are
common to all visualization in this tab:</p>
<ul>
<li>
<strong>annotation</strong>: to choose from the drop-down menu, for
example it could be the type of cell</li>
<li>
<strong>gene(s)</strong>: to type or browse from the available list,
and <strong>validate</strong> by clicking on the
<code>Validate selection</code> button. If the gene you are looking for
does not appear when you type it, it means that it does not exist in
this dataset.</li>
</ul>
<p>This tab is composed of three subtabs for various visualization:</p>
<ul>
<li><a href="#dimplot">DimPlot</a></li>
<li><a href="#vlnplot">VlnPlot</a></li>
<li><a href="#dotplot">DotPlot</a></li>
</ul>
<p>They are presented in the following paragraphs.</p>
<div class="section level3">
<h3 id="dimplot">DimPlot<a class="anchor" aria-label="anchor" href="#dimplot"></a>
</h3>
<p>The first subtab is DimPlot. This is where you finally get a view of
your dataset, in the form of a <strong>dimension plot</strong>. The type
of dimension plot is a parameter. Depending on your dataset, you might
be able to view a UMAP, t-SNE, or a PCA plot. This is a typical
representation, where each dot represent a cell.</p>
<p>The cells on the dimension plot on the right panel will have
<strong>as color the annotation</strong> parameter common to all
visualizations.</p>
<p>The cells on the dimension plot on the left panel will have as
<strong>color the gene(s) expression</strong>. The left panel will
contain as many dimension plot(s) as gene(s) selected.</p>
<p>If you do not see a plot on the left, that is because <strong>you
first need to select and validate genes</strong>.</p>
</div>
<div class="section level3">
<h3 id="vlnplot">VlnPlot<a class="anchor" aria-label="anchor" href="#vlnplot"></a>
</h3>
<p>The VlnPlot subtab presents violin plots to better understand the
<strong>distribution of gene expression</strong> across cells of
different annotations. There are as many violin plots as genes selected,
and the <strong>expression is grouped by the annotation</strong> common
to all visualizations. It can be furthered <strong>split by an
additional annotation</strong>, which is a parameter specific to this
visualization only.</p>
<p>If you do not see a plot, that is because <strong>you first need to
select and validate genes</strong>.</p>
</div>
<div class="section level3">
<h3 id="dotplot">DotPlot<a class="anchor" aria-label="anchor" href="#dotplot"></a>
</h3>
<p>The DotPlot subtab shows the <strong>average expression scaled and
percent expressed of genes</strong> (x-axis) across cells
<strong>grouped by the annotation</strong> (y-axis) common to all
visualizations.</p>
<p>If you do not see a plot, that is because <strong>you first need to
select and validate genes</strong>.</p>
</div>
</div>
<div class="section level2">
<h2 id="markers-tab">Markers tab<a class="anchor" aria-label="anchor" href="#markers-tab"></a>
</h2>
<p>The markers tab presents <strong>pre-calculated results</strong> of
Seurat’s <code>FindAllMarkers()</code>. This means that only the markers
of annotations that were found to be relevant will be available.</p>
<p>There are two elements to this tab:</p>
<ul>
<li>a heatmap</li>
<li>a table</li>
</ul>
<p>Filtering the table will update the heatmap automatically.</p>
<p>The heatmap represents the <strong>average value of the gene in all
cells from a (combination of) annotations</strong>. When applicable,
there is a possibility to split the heatmap with another (pre-chosen)
annotation.</p>
<p>The plot is interactive, browsing the tile with the mouse will show
more information, for example which annotation’s label(s) the gene is a
marker of. A gene is considered a marker of a label if the adjusted
p-value is less than 0.05 for this label in the table result of
<code>FindAllMarkers()</code>.</p>
<p>Finally, the number of genes (per annotation’s label) to show in the
heatmap is a parameter.</p>
</div>
<div class="section level2">
<h2 id="differential-expression-tab">Differential expression tab<a class="anchor" aria-label="anchor" href="#differential-expression-tab"></a>
</h2>
<p>The differential expression tab presents <strong>pre-calculated
results</strong> of Seurat’s <code>FindMarkers()</code> for specific
comparisons. This means that only the comparisons that were found to be
relevant will be available.</p>
<p>The structure of the comparison tab is very similar to the one of the
markers tab, as it also contains two elements:</p>
<ul>
<li>a heatmap</li>
<li>a table</li>
</ul>
<p>with the same characteristics.</p>
<p>The number of genes to show in the heatmap is also a parameter.</p>
</div>
<div class="section level2">
<h2 id="download-data">Download data<a class="anchor" aria-label="anchor" href="#download-data"></a>
</h2>
<p>Finally, there is a <code>Download</code> button from which you can
<strong>download the original Seurat object</strong>.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Valentine Gilbart.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
